runtime/compiler/runtime/J9Profiler.cpp

```diff
@@ -1715,9 +1715,32 @@ TR_BlockFrequencyInfo::getFrequencyInfo(
       bool normalizeForCallers,
       bool trace)
    {
-   int64_t maxCount = normalizeForCallers ? getMaxRawCount() : getMaxRawCount(bci.getCallerIndex());
    int32_t callerIndex = bci.getCallerIndex();
-   int32_t frequency = getRawCount(callerIndex < 0 ? comp->getMethodSymbol() : comp->getInlinedResolvedMethodSymbol(callerIndex), bci, _callSiteInfo, maxCount, comp);
+   int32_t queriedCallerIndex = callerIndex;
+   // Checking if the inlined method at caller Index from the frequency data is
```

Comment:
@jdmpapin  Here is the logic that checks if the method at the caller index from both current compilation and in the profiling data matches and if not, this will check if we do have the exact same call chain but at different caller index.